spec-0007/index.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "SPEC 7 — Seeding pseudo-random number generation"
+date: 2023-04-19
+author:
+  - Other participants in the discussion <not.yet@named.org>"
+  - "Stéfan van der Walt <stefanv@berkeley.edu>"
+discussion: https://github.com/scipy/scipy/issues/14322
+endorsed-by:
+---
+
+## Description
+
+<!--
+Briefly and clearly describe the proposal.
+Explain the general need and the advantages of this specific proposal.
+If relevant, include examples of how the new functionality would be used,
+intended use-cases, and pseudo-code illustrating its use.
+-->
+
+There is disparity in the APIs libraries use to seed random number generation.
+The goal of this SPEC is suggest a single, pragmatic API for the ecosystem, taking into account technical and historical factors.
+
+### Concepts
+
+- `BitGenerator`: Generates a stream of pseudo-random bits. The default generator in NumPy (`np.random.default_rng`) uses PCG64.
+- `Generator`: Derives pseudo-random numbers from the bits produced by a `BitGenerator`.
+- `RandomState`: a [legacy object in NumPy](https://numpy.org/doc/stable/reference/random/index.html), similar to `Generator`, that produces random numbers based on the Mersenne Twister.
+
+### Constraints
+
+NumPy, SciPy, scikit-learn, scikit-image, and NetworkX all implement pseudo-random seeding in slightly different ways.
+Common keyword arguments include `random_state` and `seed`.
+In practice, the seed is unfortunately also often controlled using `np.random.seed`.
+
+## Implementation
+
+<!--
+Discuss how this would be implemented.
+-->
+
+Legacy behavior in packages such as scikit-learn (`sklearn.utils.check_random_state`) typically handle `None` (use the global seed state), an int (convert to `RandomState`), or `RandomState` object.
+
+Two strong motivations for moving over to `Generator`s are:
+
+(1) it avoids naïve seeding strategies, such as using successive integers, via the underlying [SeedSequence](https://numpy.org/doc/stable/reference/random/parallel.html#seedsequence-spawning);
+(2) it avoids using global state for seeding.
```

Comment:
```suggestion

(1) it allows safe and powerful parallel streams, via the underlying [SeedSequence](https://numpy.org/doc/stable/reference/random/parallel.html#seedsequence-spawning);
(2) it avoids creating implicit global state.
```

Not sure how to rewrite (2) in more positive wording. "for seeding" was a distractor there. We're really referring to the implicit global state of `np.random.mtrand._rand` that is used by the `np.random.random()` et al. convenience functions, and the benefit of the new framework is really that we just don't do that anymore. It fundamentally _is_ something that we "don't do" anymore rather than something that we "do better".